cryptographic_estimators/UOVEstimator/uov_algorithm.py

```diff
@@ -0,0 +1,40 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..base_algorithm import BaseAlgorithm
+from .uov_problem import UOVProblem
+
+
+class UOVAlgorithm(BaseAlgorithm):
+    def __init__(self, problem: UOVProblem, **kwargs):
+        """
+        Base class for UOV algorithms complexity estimator
+
+        INPUT:
+
+        - ``problem`` -- UOVProblem object including all necessary parameters
```

Comment:
Please specify and consider the following inputs (see MQ algorithm for reference):
```        - ``w`` -- linear algebra constant (default: 2)
        - ``h`` -- external hybridization parameter (default: 0)
        - ``theta`` -- exponent of the conversion factor (default: 2)
        - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
        - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
        
        ```

Reply:
Should we include error handling for `w` and `h` variables ?

Reply:
yes, similarly as in the `MQEstimator`